internal/pkg/agent/application/gateway/fleet/fleet_gateway_test.go

```diff
@@ -564,3 +565,74 @@ func TestAgentStateToString(t *testing.T) {
 		})
 	}
 }
+
+func TestFleetGatewaySchedulerSwitch(t *testing.T) {
+	agentInfo := &testAgentInfo{}
+	settings := &fleetGatewaySettings{
+		Duration: 5 * time.Second,
+		Backoff:  backoffSettings{Init: 1 * time.Second, Max: 5 * time.Second},
+	}
+
+	t.Run("if unauthorized responses exceed the set limit, the scheduler should be switched to the long-wait scheduler", withGateway(agentInfo, settings, func(
+		t *testing.T,
+		gateway coordinator.FleetGateway,
+		c *testingClient,
+		sch *scheduler.Stepper,
+	) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		unauth := func(_ http.Header, _ io.Reader) (*http.Response, error) {
+			return nil, client.ErrInvalidAPIKey
+		}
+
+		clientWaitFn := c.Answer(unauth)
+		g, ok := gateway.(*FleetGateway)
+		require.True(t, ok)
+		g.scheduler = scheduler.NewPeriodicJitter(time.Second, time.Second)
+
+		errCh := runFleetGateway(ctx, gateway)
+
+		for i := 0; i <= maxUnauthCounter; i++ {
+			<-clientWaitFn
+		}
+
+		cancel()
+		err := <-errCh
+		require.NoError(t, err)
+
+		_, ok = g.scheduler.(*scheduler.Periodic)
```

Comment:
This only checks that the type of the scheduler changed not that the retry interval actually changed as expected. That is the behavior you want to test for.

Reply:
This test would still pass if the retry duration got shorter for example.